BLUESPAWN-client/src/util/filesystem/FileSystem.cpp

```diff
@@ -612,6 +611,24 @@ namespace FileSystem{
 		return this->SetFileOwner(*BluespawnOwner);
 	}
 
+	bool File::GrantPermissions(const Permissions::Owner& owner, const ACCESS_MASK& amAccess) {
+		return Permissions::UpdateObjectACL(FilePath, SE_FILE_OBJECT, owner, amAccess);
+	}
+
+	bool File::DenyPermissions(const Permissions::Owner& owner, const ACCESS_MASK& amAccess) {
+		return Permissions::UpdateObjectACL(FilePath, SE_FILE_OBJECT, owner, amAccess, true);
+	}
+
+	bool File::Quarantine() {
+		if (!bFileExists) {
+			LOG_ERROR("Can't quarantine file " << FilePath << ". File doesn't exist");

```

Comment:
SetLastError(ERROR_FILE_NOT_FOUND)

Reply:
since you print out the error if one occurs here, it's good to make sure it reflects what happened.